### PR TITLE
 Validate parameters schema in Provision/Update/Bind request body

### PR DESCRIPTION
**Pull request by leonwanghui**
Sloves #44

### CODE REVIEW

File: 2.13/tests/test/test.js [hunk 1]
```diff
@@ -194,12 +199,25 @@ function testProvision (instanceId, validBody, isAsync) {
...
+              .put('/v2/service_instances/' + instanceId + '?accepts_incomplete=true')
+              .set('X-Broker-API-Version', apiVersion)
+              .auth(config.user, config.password)
+              .send(tempBody)
+              .expect(400, done)
+          }
+          callback(null)
+        }
+      ], function (err) {
+        if (err) { return done(err) } else { done() }
```
Comment (zhongyi-zhang) [inline 2.13/tests/test/test.js]: Can be simple `done(err)`. Same in several places below.

File: 2.13/tests/test/test.js [hunk 2]
```diff
@@ -655,3 +723,71 @@ function validateJsonSchema (body, schema) {
...
+    for (var i in obj) {
+      var found = findWhichContains(obj[i], key, value)
+      if (found) return found
+    }
+  } else if (typeof obj === 'object') {
+    for (var p in obj) {
+      if (p === key && obj[p] === value) {
+        return obj
+      }
+      found = findWhichContains(p, key, value)
```
Comment (norshtein) [inline 2.13/tests/test/test.js]: Should be `findWhichContains(obj[p], key, value)`. Also, you may want to rename `p` to `k` to indicate it is a "key".

File: 2.13/tests/test/test.js [hunk 3]
```diff
@@ -655,3 +723,71 @@ function validateJsonSchema (body, schema) {
...
+    return obj
+  }
+  if (Array.isArray(obj)) {
+    for (var i in obj) {
+      var found = findWhichContains(obj[i], key, value)
+      if (found) return found
+    }
+  } else if (typeof obj === 'object') {
+    for (var p in obj) {
+      if (p === key && obj[p] === value) {
```
Comment (norshtein) [inline 2.13/tests/test/test.js]: A redundant test since you have tested it in L753-755

File: 2.13/tests/test/test.js (hunk 2 above)
Comment (leonwanghui) [reply 2.13/tests/test/test.js]: Got it, I think we should also update it in mock server.

**Pull request merged**